scripts/pre_push_hook.py

```diff
@@ -453,6 +455,16 @@ def main(args=None):
                     'Push aborted due to failing typescript checks.')
                 sys.exit(1)
 
+            strict_typescript_checks_status = 0
+            if does_diff_include_ts_files(files_to_lint):
+                strict_typescript_checks_status = run_script_and_get_returncode(
+                    STRICT_TYPESCRIPT_CHECKS_CMDS)
```

Comment:
Do we need to run the typescript checks twice? If yes, will it be possible to run the "normal/simple" checks as well using the --strict_mode?

Also, if we are running the script twice are we compiling files twice?

Reply:
Actually the thing is that we want to make the typescript checks more strict so we are enabling the `strict` flag in tsconfig. But the strict checks can't be enabled in the complete codebase due to a lot of ts errors. You can refer the discussion at #10327 

So, only some files are compiled using the strict config and we plan to cover the complete codebase gradually.

Also, I think it would be better to run only the strict checks when `--strict_checks` flag is used. 

The time added to the pre_push_hooks is about 3 seconds.

Reply:
I see, so in the future, we will only have strict checks. Thanks for the info!

Reply:
Yup, correct!